src/WorkflowCore/Services/DefaultProviders/SingleNodeEventHub.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using WorkflowCore.Interface;
+using WorkflowCore.Models.LifeCycleEvents;
+
+namespace WorkflowCore.Services
+{
+    public class SingleNodeEventHub : ILifeCycleEventHub
+    {
+        private ICollection<Action<LifeCycleEvent>> _subscribers = new HashSet<Action<LifeCycleEvent>>();
+        private readonly ILogger _logger;
+
+        public SingleNodeEventHub(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<SingleNodeEventHub>();
+        }
+
+        public Task PublishNotification(LifeCycleEvent evt)
+        {
+            Task.Run(() =>
+            {
+                foreach (var subscriber in _subscribers)
+                {
+                    try
+                    {
+                        subscriber(evt);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogWarning($"Error on event subscriber: {ex.Message}");
```

Comment:
```suggestion
                        _logger.LogWarning(ex, $"Error on event subscriber: {ex.Message}");
```

can you log exception here so we have the stack trace if anything goes wrong?